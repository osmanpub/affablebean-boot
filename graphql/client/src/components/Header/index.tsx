import React from "react";
import { Link } from "react-router-dom";
import WidgetBar from "../WidgetBar";
import "./Header.css";
import { HeaderMain, HeaderWrapper, Logo } from "./Header.styles";

type Props = {
  url: string;
};

export default function Header(props: Props) {
  const { url } = props;

  return (
    <HeaderWrapper>
      <HeaderMain>
        <WidgetBar url={url} />
        <Logo>
          <Link to={"/"} data-cy="home-logo">
            <img src="/static/img/logo.png" alt="Affable Bean logo" />
          </Link>
        </Logo>
        <img
          alt="the affable bean"
          id="logoText"
          src="/static/img/logoText.png"
        />
      </HeaderMain>
    </HeaderWrapper>
  );
}
