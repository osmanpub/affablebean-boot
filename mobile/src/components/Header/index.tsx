import React from 'react';
import {
  Image,
  StyleSheet,
  Text,
  TouchableHighlight,
  TouchableWithoutFeedback,
  View,
} from 'react-native';
import {connect} from 'react-redux';
import {Cart} from '../../interfaces/cart';
import {RootState} from '../../redux';

type Props = {
  cart: Cart;
  currentScreen: string;
  setScreen: Function;
};

function Header(props: Props) {
  const {cart, currentScreen, setScreen} = props;

  let cartWidget = null;
  let checkoutWidget = null;

  if (cart && cart.numberOfItems > 0) {
    if (currentScreen !== 'Cart') {
      cartWidget = (
        <TouchableHighlight onPress={() => setScreen('Cart')}>
          <Text style={styles.checkout}>view cart</Text>
        </TouchableHighlight>
      );
    }

    if (currentScreen !== 'Checkout') {
      checkoutWidget = (
        <TouchableWithoutFeedback onPress={() => setScreen('Checkout')}>
          <Text style={styles.viewCart}>checkout</Text>
        </TouchableWithoutFeedback>
      );
    }
  }

  return (
    <View
      style={{
        alignItems: 'center',
      }}>
      <TouchableWithoutFeedback onPress={() => setScreen('Home')}>
        <Image source={require('./logo.jpg')} style={styles.logo} />
      </TouchableWithoutFeedback>

      {cart && cart.numberOfItems > 0 && (
        <View style={{flexDirection: 'row'}}>
          {cartWidget}
          {checkoutWidget}
          <View style={{flexDirection: 'row', padding: 8}}>
            <Image source={require('./cart.gif')} />
            <Text>&nbsp;{cart.numberOfItems}&nbsp;items</Text>
          </View>
        </View>
      )}
    </View>
  );
}

const styles = StyleSheet.create({
  checkout: {
    borderColor: 'blue',
    borderRadius: 32,
    borderWidth: 1,
    color: 'blue',
    fontWeight: 'bold',
    padding: 8,
  },
  logo: {
    margin: 16,
  },
  viewCart: {
    borderColor: 'blue',
    borderRadius: 32,
    borderWidth: 1,
    color: 'blue',
    fontWeight: 'bold',
    marginBottom: 16,
    marginLeft: 16,
    marginRight: 16,
    padding: 8,
  },
});

const mapStateToProps = (state: RootState) => ({
  cart: state.cart,
});

export default connect(mapStateToProps)(Header);
