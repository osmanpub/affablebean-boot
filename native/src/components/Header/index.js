import React from "react";
import {
  Image,
  StyleSheet,
  Text,
  TouchableHighlight,
  View
} from "react-native";
import PropTypes from "prop-types";

export default function Header(props) {
  const { cart, currentScreen, setScreen } = props;

  const styles = StyleSheet.create({
    checkout: {
      borderColor: "blue",
      borderRadius: 32,
      borderWidth: 1,
      color: "blue",
      fontWeight: "bold",
      padding: 8
    },
    logo: {
      margin: 16
    },
    viewCart: {
      borderColor: "blue",
      borderRadius: 32,
      borderWidth: 1,
      color: "blue",
      fontWeight: "bold",
      marginBottom: 16,
      marginLeft: 16,
      marginRight: 16,
      padding: 8
    }
  });

  function goHome() {
    setScreen("Home");
  }

  function viewCart() {
    if (currentScreen !== "Cart") {
      setScreen("Cart");
    }
  }

  function viewCheckout() {
    if (currentScreen !== "Checkout") {
      setScreen("Checkout");
    }
  }

  return (
    <View
      style={{
        alignItems: "center",
        background: "#f7f7e9"
      }}
    >
      <TouchableHighlight onPress={goHome}>
        <Image source={require("./logo.jpg")} style={styles.logo} />
      </TouchableHighlight>

      {cart && cart.numberOfItems > 0 && (
        <View style={{ flexDirection: "row" }}>
          <TouchableHighlight onPress={viewCart}>
            <Text style={styles.checkout}>view cart</Text>
          </TouchableHighlight>
          <TouchableHighlight onPress={viewCheckout}>
            <Text style={styles.viewCart}>checkout</Text>
          </TouchableHighlight>
          <View style={{ flexDirection: "row", padding: 8 }}>
            <Image source={require("./cart.gif")} />
            <Text>&nbsp;{cart.numberOfItems}&nbsp;items</Text>
          </View>
        </View>
      )}
    </View>
  );
}

Header.propTypes = {
  cart: PropTypes.object,
  currentScreen: PropTypes.string.isRequired,
  setScreen: PropTypes.func.isRequired
};
