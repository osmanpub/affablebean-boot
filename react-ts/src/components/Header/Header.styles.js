import styled from "styled-components";

export const HeaderWrapper = styled.div`
  background: #f7f7e9;
`;

export const HeaderMain = styled.div`
  width: 1080px;
  height: 250px;
  background: url(/static/img/stalk.png) no-repeat 28px 28px;
`;

export const Logo = styled.div`
  float: left;
  margin-left: 30px;
  margin-top: -20px;
`;
