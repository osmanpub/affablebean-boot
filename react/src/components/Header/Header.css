#logoText {
  float: left;
  margin: 20px 0 0 70px;
  /* font styles apply to text within img alt attribute */
  font-family: "American Typewriter", Courier, monospace;
  font-size: 50px;
  color: #333;
}
