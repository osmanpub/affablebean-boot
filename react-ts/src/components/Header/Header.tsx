import React from "react";
import { Link } from "react-router-dom";
import { HeaderMain, HeaderWrapper, Logo } from "./Header.styles";
import WidgetBar from "../WidgetBar";
import "./Header.css";
import { HeaderProps } from "../../interfaces/props";

export function Header(props: HeaderProps) {
  const { cart, url } = props;

  return (
    <HeaderWrapper>
      <HeaderMain>
        <WidgetBar cart={cart} url={url} />
        <Logo>
          <Link to={"/"}>
            <img src="/static/img/logo.png" alt="Affable Bean logo" />
          </Link>
        </Logo>
        <img
          id="logoText"
          src="/static/img/logoText.png"
          alt="the affable bean"
        />
      </HeaderMain>
    </HeaderWrapper>
  );
}
