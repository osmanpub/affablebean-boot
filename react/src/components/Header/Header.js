import React from "react";
import PropTypes from "prop-types";
import { Link } from "react-router-dom";
import { HeaderMain, HeaderWrapper, Logo } from "./Header.styles";
import WidgetBar from "../WidgetBar";
import "./Header.css";

export function Header(props) {
  const { cart, url } = props;

  return (
    <HeaderWrapper>
      <HeaderMain>
        <WidgetBar cart={cart} url={url} />
        <Logo>
          <Link to={"/"}>
            <img src="/static/img/logo.png" alt="Affable Bean logo" />
          </Link>
        </Logo>
        <img
          id="logoText"
          src="/static/img/logoText.png"
          alt="the affable bean"
        />
      </HeaderMain>
    </HeaderWrapper>
  );
}

Header.propTypes = {
  cart: PropTypes.object,
  url: PropTypes.string.isRequired
};
