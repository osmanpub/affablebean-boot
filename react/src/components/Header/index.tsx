import React from "react";
import { Link } from "react-router-dom";
import WidgetBar from "../WidgetBar";
import "./Header.css";
import { HeaderMain, HeaderWrapper, Logo } from "./Header.styles";

type Props = {
  url: string;
};

export default function Header(props: Props) {
  const { url } = props;

  return (
    <HeaderWrapper>
      <HeaderMain>
        <WidgetBar url={url} />
        <Logo>
          <Link to={"/"}>
            <img src="/static/img/logo.png" alt="Affable Bean logo" />
          </Link>
        </Logo>
        <img
          id="logoText"
          src="/static/img/logoText.png"
          alt="the affable bean"
        />
      </HeaderMain>
    </HeaderWrapper>
  );
}
