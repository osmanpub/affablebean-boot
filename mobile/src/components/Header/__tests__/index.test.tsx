import {render} from '@testing-library/react-native';
import React from 'react';
import {Provider} from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import Header from '..';

const mockStore = configureStore([]);

const initialState = {
  cart: {
    didInvalidate: false,
    isFetching: false,
    items: [
      {
        product: {
          id: 1,
          description: 'semi skimmed (1L)',
          name: 'milk',
          price: 1.7,
        },
        quantity: 1,
        total: 1.7,
      },
    ],
    numberOfItems: 1,
    subtotal: 1.7,
  },
};

const store = mockStore(initialState);

const header = (
  <Provider store={store}>
    <Header currentScreen="" setScreen={jest.fn()} />
  </Provider>
);

describe('<Header />', () => {
  it('renders correctly', () => {
    const component = renderer.create(header).toJSON();
    expect(component).toMatchSnapshot();
  });

  it('show logo', () => {
    const {getByTestId} = render(header);
    const logo = getByTestId(/logo/i);
    expect(logo).toBeTruthy();
  });

  it('show view cart link', () => {
    const {getByTestId} = render(header);
    const viewCart = getByTestId(/view-cart/i);
    expect(viewCart).toBeTruthy();
  });

  it('show clear cart link', () => {
    const {getByTestId} = render(header);
    const clearCart = getByTestId(/clear-cart/i);
    expect(clearCart).toBeTruthy();
  });

  it('show checkout link', () => {
    const {getByTestId} = render(header);
    const checkout = getByTestId(/checkout/i);
    expect(checkout).toBeTruthy();
  });

  it('show 1 cart items', () => {
    const {getByText} = render(header);
    const items = getByText(/1 items/i);
    expect(items).toBeTruthy();
  });
});
