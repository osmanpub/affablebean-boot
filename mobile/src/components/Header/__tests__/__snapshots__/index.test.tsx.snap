// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
    }
  }
>
  <TouchableWithoutFeedback
    onPress={[Function]}
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/components/Header/logo.jpg",
        }
      }
      style={
        Object {
          "margin": 16,
        }
      }
      testID="logo"
    />
  </TouchableWithoutFeedback>
  <View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "paddingBottom": 16,
        }
      }
    >
      <TouchableHighlight
        activeOpacity={0.85}
        delayPressOut={100}
        onPress={[Function]}
        underlayColor="black"
      >
        <Text
          style={
            Object {
              "borderColor": "blue",
              "borderRadius": 32,
              "borderWidth": 1,
              "color": "blue",
              "fontWeight": "bold",
              "marginHorizontal": 8,
              "padding": 12,
            }
          }
          testID="clear-cart"
        >
          clear cart
        </Text>
      </TouchableHighlight>
      <TouchableHighlight
        activeOpacity={0.85}
        delayPressOut={100}
        onPress={[Function]}
        underlayColor="black"
      >
        <Text
          style={
            Object {
              "borderColor": "blue",
              "borderRadius": 32,
              "borderWidth": 1,
              "color": "blue",
              "fontWeight": "bold",
              "marginHorizontal": 8,
              "padding": 12,
            }
          }
          testID="view-cart"
        >
          view cart
        </Text>
      </TouchableHighlight>
      <TouchableWithoutFeedback
        onPress={[Function]}
      >
        <Text
          style={
            Object {
              "borderColor": "blue",
              "borderRadius": 32,
              "borderWidth": 1,
              "color": "blue",
              "fontWeight": "bold",
              "marginHorizontal": 8,
              "padding": 12,
            }
          }
          testID="checkout"
        >
          checkout
        </Text>
      </TouchableWithoutFeedback>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
          "paddingBottom": 16,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/components/Header/cart.gif",
          }
        }
      />
      <Text
        testID="cart-items"
      >
         
        1
         items
      </Text>
    </View>
  </View>
</View>
`;
